### DISCUSSION TITLE
[Question] Coder Server didn't show icon via HTTP protocol

### DISCUSSION THREAD

**Discussion by lnvn** [coder/code-server#6997] [Q&A] [answered]
Hi all, when I tried to deploy coder-server in an AWS EC2 instance via this command:
`curl -fsSL https://code-server.dev/install.sh | sh`

and configured:
```
$ cat config.yaml
bind-addr: 0.0.0.0:8080
auth: password
password: mypass
```

then I access the code server via my http://my-ip:8080/ but it did not show icon on the console

<img width="178" alt="coder-feature" src="https://github.com/user-attachments/assets/ec26a05d-f0fa-4af9-8e6b-2f015715c127">

Any body have a suggestion to fix this issue?
Thanks a lot!

**Comment by code-asher** [comment]
Are there any error messages in the browser console or failed requests in the network tab?

**Reply by lnvn** [reply]
Hi @code-asher,

I see some err **in console log**
```
webcontent.js:1 Uncaught TypeError: Cannot read properties of undefined (reading 'browserObject')
    at webcontent.js:1:264
    at webcontent.js:3:152
    at webcontent.js:3:156
```

```
Failed to decode downloaded font: <URL>
  Failed to decode downloaded font: http://my-ip:8080/stable-34c6751bf8ae4e343f908d8aee691a4778f1d3a2/static/out/vs/base/browser/ui/codicons/codicon/codicon.ttf?5d4d76ab2ce5108968ad644d591a16a6
  Failed to decode downloaded font: http://my-ip:8080/stable-34c6751bf8ae4e343f908d8aee691a4778f1d3a2/static/out/vs/base/browser/ui/codicons/codicon/codicon.ttf?5d4d76ab2ce5108968ad644d591a16a6
...


OTS parsing error: invalid sfntVersion: 1013478509
...
```

```
GET http://my-ip:8080/stable-34c6751bf8ae4e343f908d8aee691a4778f1d3a2/static/out/vsda.js net::ERR_ABORTED 404 (Not Found)
```

and in **Network** tab, the request to vsda.js got 404 Not Found

**Reply by code-asher** [reply, marked as answer]
No need to worry about `vsda.js`, that is not available in the open source version of VS Code.

The `browserObject` error is interesting.  What browser and version are you using?

But the font error is definitely the one causing the issue.  Are you able to directly open `http://my-ip:8080/stable-34c6751bf8ae4e343f908d8aee691a4778f1d3a2/static/out/vs/base/browser/ui/codicons/codicon/codicon.ttf?5d4d76ab2ce5108968ad644d591a16a6` and download the font?  Or does it show HTML or something instead?

**Reply by lnvn** [reply]
@code-asher I can not open "http://my-ip:8080/stable-34c6751bf8ae4e343f908d8aee691a4778f1d3a2/static/out/vs/base/browser/ui/codicons/codicon/codicon.ttf?5d4d76ab2ce5108968ad644d591a16a6" **because of the network policy**

I think we know the problem here, thanks a lot @code-asher for help me to point out the reason.
